pkg/controller/conditions/conditions.go

```diff
@@ -36,12 +35,14 @@ func SetCondition(instance *nmstatev1.NodeNetworkState, conditionType nmstatev1.
 	}
 
 	condition.LastHeartbeatTime = now
+
+	return conditions
 }
 
-func Condition(instance *nmstatev1.NodeNetworkState, conditionType nmstatev1.NodeNetworkStateConditionType) *nmstatev1.NodeNetworkStateCondition {
-	for i, condition := range instance.Status.Conditions {
+func Condition(conditions nmstatev1.ConditionList, conditionType nmstatev1.ConditionType) *nmstatev1.Condition {
```

Comment:
Ask @qinqon, he would link you to Go style guideline saying that instead of getters, we should use property-like names. In this case though, since it is not a property (it takes a lot of args), we may change it. What do you think, Quique?

I see, will change it to not-a-pointer then.

Reply:
Yes but `Condition` returning a `Condition` object makes me thing to a constructor. Here what the function does is to find the element inside the list.

Reply:
Arriving late to this, but I totally agree with @fedepaol a method with ConditionList as receiver is the way to go, and use a verb for it Find is good if we end with more than one find then we do the FindByFoo

About getters style, is from effective go https://golang.org/doc/effective_go.html#Getters